web/src/lib/components/settings/Nostr.svelte

```diff
@@ -0,0 +1,78 @@
+<script lang="ts">
+    import {user} from "$lib/stores";
+    import {getPublicKey} from "nostr-tools";
+    import {onMount} from "svelte";
+    import {hasExtension} from "$lib/nostr/utils";
+
+    let nostr_public_key = getPublicKey($user.nostr_private_key);
+    let nostrExtensionEnabled: boolean;
+
+    onMount(async () => {
+        nostrExtensionEnabled = hasExtension();
+    })
+</script>
+
+<div class="text-2xl breadcrumbs">
+    <ul>
+        <li>Settings</li>
+        <li>Nostr</li>
+    </ul>
+</div>
+
+<div class="mt-8">
+    {#if !nostrExtensionEnabled}
+        <div class="alert alert-success shadow-lg">
+            <div>
+                <svg xmlns="http://www.w3.org/2000/svg" class="stroke-current flex-shrink-0 h-6 w-6" fill="none" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M9 12l2 2 4-4m6 2a9 9 0 11-18 0 9 9 0 0118 0z" /></svg>
+                <span>We have detected that you have a Nostr extension in your browser, so you'll be able to use your
+                own identity in the entire Plebeian Market.</span>
+            </div>
+        </div>
+
+    {:else}
+        <div class="alert alert-warning shadow-lg">
+            <div>
+                <svg xmlns="http://www.w3.org/2000/svg" class="stroke-current flex-shrink-0 h-6 w-6" fill="none" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 9v2m0 4h.01m-6.938 4h13.856c1.54 0 2.502-1.667 1.732-3L13.732 4c-.77-1.333-2.694-1.333-3.464 0L3.34 16c-.77 1.333.192 3 1.732 3z" /></svg>
+                <span>You don't have a Nostr extension</span>
+            </div>
+        </div>
+
+        <div class="mt-4 text-justify">
+            <p>You're using your Plebeian Market generated Nostr identity. We created this for you as we detected
```

Comment:
"so nobody knows your private key" not technically correct, also sounds a bit strange. Maybe "so nobody else has access to your private key"?